cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -99,93 +83,54 @@ function wait_k8s_object {
-  echo $error_message && return  1
-}
-
-
-function is_taint_absence {
-  local -r taint=$1
-
-  result=$(_kubectl get nodes -o jsonpath="{.items[*].spec.taints[?(@.effect == \"$taint\")].effect}" || echo error)
-  if [[ -z $result ]]; then
-    echo "not-present"
+  if ! retry "$tries" "$wait_time" "$action" "$wait_message";then 
+    echo $error_message
+    return  1
   fi
-}
-
-function wait_for_taint_absence {
-  local -r taint=$1
 
-  local -r tries=60
-  local -r wait_time=5
-
-  local -r wait_message="Waiting for $taint taint absence"
-  local -r error_message="Taint $taint $name did not removed"
-  local -r action="is_taint_absence $taint"
-
-  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
-  echo $error_message && return 1
-}
-
-function wait_for_taint {
-  local -r taint=$1
-
-  local -r tries=60
-  local -r wait_time=5
-
-  local -r wait_message="Waiting for $taint taint to present"
-  local -r error_message="Taint $taint $name did not present"
-  local -r action="_kubectl get nodes -o custom-columns=taints:.spec.taints[*].effect --no-headers | grep -i $taint"
-
-  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
-  echo $error_message && return 1
+  return 0
 }
 
 # not using kubectl wait since with the sriov operator the pods get restarted a couple of times and this is
 # more reliable
 function wait_pods_ready {
-    while [ -n "$(_kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers | grep false)" ]; do
-        echo "Waiting for all pods to become ready ..."
-        _kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
-        sleep 10
-    done
-}
-
-function wait_pods_condition_by_label {
-  local -r tries=10
-  local -r wait_time=1
-  local -r namespace=$1
-  local -r label=$2
-  local -r condition=$3
+  local -r tries=30
+  local -r wait_time=10
 
-  local -r wait_message="waiting for pod label $label at namespace $namespace to meet condition $condition"
-  local -r error_message="pod label $label at namespace $namespace did not meet condition $condition"
-  local -r action="_kubectl wait pods -n $namespace -l $label --for condition=$condition --timeout 30s >/dev/null && echo succeed"
+  local -r wait_message="Waiting for all pods to become ready.."
+  local -r error_message="Not all pods were ready after $(($tries*$wait_time)) seconds"
+  local -r get_pods_status="cluster-up/kubectl.sh get pods --all-namespaces -ocustom-columns=status:status.conditions[?(@.type==\"Ready\")].status,metadata:metadata.name --no-headers"
+  local -r action='[[ -z $($get_pods_status | grep -i false) ]] && echo all-pods-ready'
 
-  if ! retry "$tries" "$wait_time" "$action" "$wait_message"; then
+  if ! retry "$tries" "$wait_time" "$action" "$wait_message" "$get_pods_status"; then
     echo $error_message
     return 1
   fi
+  
+  return 0
```

Comment:
Redundant, all over, without explicit return, the function will return the ret code of the last command (which is not "hidden").
Unless you want to be explicit.

Reply:
I prefer it to be explicit so in case one of the operations inside the function will print a message to STDOUT
the function will return a value that is only 1 or 0

Reply:
if will surely return 0, otherwise the `if` above it would return an error
```
#/bin/bash -ex

foo ()
{
  echo bla
  false
  if ! true; then
    echo 11
  fi
}

foo
echo $?
```
This one return 0, as it look on the last command, even that it had `false` above it.

Reply:
I prefer to stay explicit in this script helper functions at least for functions that use `retry`

Reply:
It break the one liner approach which was suggested by Quique in the same context imo.
there is a different between explicit and redundant, it might be subjective.

Reply:
Quique asked to remove the conditional return one liners because they were prune to errors.
We already in a script so we dont benefit much from one liners

Reply:
I am talking about https://github.com/kubevirt/kubevirtci/pull/481#discussion_r529459786

Reply:
Yeah I am familiar with that comment Quique asked for the same on this PR 
I understand what you see it redundant , I mean that since we use `return 1` we should also use `return 0` to be explicit